src/json_node.rs

```diff
@@ -0,0 +1,240 @@
+use crate::select::select_value::{SelectValue, SelectValueType, ValueUpdater};
+use crate::select::JsonPathError;
+use crate::select::SelectorMut;
```

Comment:
It's hard to track exactly what you changed as compared to the original `jsonpath_lib`. I did a manual diff, but that's inconvenient compared to Git.

Instead, it would be better to keep the crate separate by forking the original and using the fork here (this is what we originally did). That way it will be easy to see what has changed, and also to track future upstream changes.
It will also be cleaner, keeping the code separate. And last but not least, the build will be faster.

Reply:
I agree, will make a PR to our fork and you can review it there ..